File: smartadapter/src/main/java/smartadapter/state/SmartStateHolder.java
package smartadapter.state;

/*
 * Created by Manne Öhlund on 2019-08-09.
 * Copyright (c) All rights reserved.
 */

/**
 * Defines the SortedSet for enabled adapter positions.
 */
public interface SmartStateHolder {

    /**
     * Adds the position to the data set.
     * @param position the adapter position
     */
    void enable(int position);

    /**
     * Removes the position from the data set.
     * @param position the adapter position
     */
    void disable(int position);

    /**
     * Enables or disables the position in the data set.
     * @param position the adapter position
     */
    void toggle(int position);

    /**
     * Clears all the stored states.
     */
    void clear();
}

File: smartadapter/src/main/java/smartadapter/state/SmartStateHolder.kt
package smartadapter.state

import smartadapter.Position

/*
 * Created by Manne Öhlund on 2019-08-09.
 * Copyright (c) All rights reserved.
 */

/**
 * Defines the SortedSet for enabled adapter positions.
 */
interface SmartStateHolder {

    /**
     * Adds the position to the data set.
     * @param position the adapter position
     */
    fun enable(position: Position)

    /**
     * Removes the position from the data set.
     * @param position the adapter position
     */
    fun disable(position: Position)

    /**
     * Enables or disables the position in the data set.
     * @param position the adapter position
     */
    fun toggle(position: Position)

    /**
     * Clears all the stored states.
     */
    fun clear()
}
